internal/services/notification/runwebhookdelivery.go

```diff
@@ -0,0 +1,203 @@
+// Copyright 2023 Sorint.lab
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package notification
+
+import (
+	"bytes"
+	"context"
+	"crypto/hmac"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"net/http"
+	"time"
+
+	"github.com/sorintlab/errors"
+
+	"agola.io/agola/internal/sqlg/lock"
+	"agola.io/agola/internal/sqlg/sql"
+	"agola.io/agola/internal/util"
+	"agola.io/agola/services/notification/types"
+)
+
+const (
+	MaxRunWebhookDeliveriesLimit = 40
+	RunWebhookDeliveriesLockKey  = "runwebhookdeliveryevents"
+
+	// runWebhookDeliveriesInterval is the maximum time to wait for run webhook delivery timeout handler
```

Comment:
// runWebhookDeliveriesInterval is the time to wait between every runWebhookDeliveriesHandler call.